mixed-content/generic/worker.js

```diff
@@ -0,0 +1,7 @@
+try {
+  // Works on same origin.
+  postMessage('done');
+} catch(err) {
+  // Works cross-origin.
+  postMessage('done', '*');
```

Comment:
Cross origin is HTTPS --> HTTP.

We can examine it further, if you like, I can extract out just those tests to show what the issue is, at least in Chrome. Might be worthwhile investigating this further. :-)